### ISSUE TITLE
time_to_freq error with field with multiple components

### ISSUE THREAD

**Issue by SebGue** [Eomys/SciDataTool#1]
Hello Eomys,

I get an error if I try to do a FFT with the time_to_freq method of DataTime on a field with multiple components.
If I do the FFT on a field with only one component every the transform works fine.

You can reproduce this issue with your Tutorial 1 - Example 6 (vs. Example 1) by applying time_to_freq method to Br.
By the way, the tutorials_data.xlsx file is maybe a little large and should be reduced, since I was not able to 'read_xls' the file within resonable time.

Best regards, Sebastian

**Comment by helene-t** [comment]
Hello Sebastian,

Thank you for your interest in SciDataTool :)

Indeed, the components feature is still work in progress, and has not been completely validated yet, especially with the time_to_freq method. I will have a look at it as soon as I can.

Best regards, Helene

**Comment by SebGue** [comment]
Hello Helene,

since I need the FFT transform to progress with the pyleecan iron losses, could you tell me how I can extract the field component data from the DataTime (or DataND in general) objects. (I have some workaround in my mind.)

Best regards, Sebastian

**Comment by SebGue** [comment]
Hello Helene,

i did the following workaround that seem to work. Maybe you could have a look. There may be an easier way to do this but I have not completely understood how to access data with SciDataTool.

```
#---------------------------------------------------------------
# workaround
# fourier transform with data with multiple components
# axes/components order won't be restored
#---------------------------------------------------------------

# get axes and components
axes = []
components = []
n_comp = 1
for axis in data.axes:
    if axis.is_components:
        components.append(axis)
        n_comp = len(axis.values)
    else: 
        axes.append(axis)
    
axes_str = [x.name for x in axes]

# transform each field component at a time
if n_comp > 1:
    values = []
    for id in range(n_comp):
        component_str = components[0].name + f"[{id}]"
        component_data = data.get_along(component_str,*axes_str)
        _data = DataTime(
                    name = data.name,
                    unit = data.unit,
                    symbol = data.symbol,
                    axes = axes,
                    values = component_data[data.symbol]
                    )
        
        _dataFt = _data.time_to_freq()
        values.append(_dataFt.values)

    values = array([*values])

    # recombine transformed field components
    dataFt = DataFreq(
                name=_dataFt.name,
                unit=_dataFt.unit,
                symbol=_dataFt.symbol,
                axes=[components[0], *_dataFt.axes],
                values=values
                )
else:
    dataFt = data.time_to_freq()

```

Best regards, Sebastian

**Comment by helene-t** [comment]
Hello Sebastian,

It should be easier to directly get the fft values along each component using:
```
axes_str = [x.name if x.name !="time" else "freq" for x in axes]
axes_str = [x.name if x.name !="angle" else "wavenumber" for x in axes] # if you also want a spatial fft
values = []
for id in range(n_comp):
    component_str = components[0].name + f"[{id}]"
    component_data = data.get_along(component_str,*axes_str)
    values.append(component_data[data.symbol])
values = array([*values])
```
And at the end create the dataFT as you did.
I hope to have some time next week to have a look at the issue.
Best regards, Helene

**Comment by SebGue** [comment]
Hello Helene,

I found an other issue related to time_to_freq method, i.e. it doesn't convert the unit. 
If needed, I can provide a minimal example of this issue.

 Best regards, Sebastian

**Comment by helene-t** [comment]
Hello Sebastian,

I just released a new version (1.0.6) with a patch for time_to_freq and freq_to_time in order to correctly account for components axis, and add units. Can you tell me if it fixes your issue? 

Best regards, Helene

**Comment by SebGue** [comment]
Hello Helene,

that new version seem to fix the issue. That's great. Thank you very much!

Best regards, Sebastian
